books/free-programming-books-langs.md

```diff
@@ -2223,6 +2223,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [A Pamphlet Against R. Computational Intelligence in Guile Scheme](https://panicz.github.io/pamphlet/)
 * [An Introduction to Scheme and its Implementation](http://www.cs.rpi.edu/academics/courses/fall00/ai/scheme/reference/schintro-v14/schintro_toc.html)
 * [Concrete Abstractions: An Introduction to Computer Science Using Scheme](https://gustavus.edu/+max/concrete-abstractions.html) - M. Hailperin, B. Kaiser, K. Knight
+* [Scheme 9 from Empty Space, first edition](https://web.archive.org/web/20180403032702/www.bcl.hamilton.ie/~nmh/t3x.org/zzz/scheme-9-from-empty-space.zip) - Nils M. Holm
```

Comment:
- Update with latest archived url, notes included.
- Adding year for historical resources
- Adding formats
- Link to interpreter download compatible with those sources.

```suggestion
* [Scheme 9 from Empty Space - First edition (2008)](https://web.archive.org/web/20181010075503/http://www.bcl.hamilton.ie/~nmh/t3x.org/zzz/scheme-9-from-empty-space.zip) - Nils M. Holm (PDF, :package: sources, [:cd: interpreter](https://web.archive.org/web/20161206133212/http://www.bcl.hamilton.ie/~nmh/t3x.org/zzz/s9fes-20090906.tar.gz)) *(:card_file_box: archived)*
```

There are not any other format than zip

Reply:
Something like this?

Reply:
> `[:cd: interpreter](https://web.archive.org/web/20161206133212/http://www.bcl.hamilton.ie/~nmh/t3x.org/zzz/s9fes-20090906.tar.gz)) *(:card_file_box: archived)*`

The source code in this tarball is not the same as the code shown in the book.

Reply:
Isn't the copyright year 2007 instead of 2008?

Reply:
> > `[:cd: interpreter](https://web.archive.org/web/20161206133212/http://www.bcl.hamilton.ie/~nmh/t3x.org/zzz/s9fes-20090906.tar.gz)`
> 
> The source code in this tarball is not the same as the code shown in the book.

This is the other downloable related with the book: the `R4RS Scheme interpreter` matching same archived date

See: https://web.archive.org/web/20181010075503/https://www.bcl.hamilton.ie/~nmh/t3x.org/zzz/

I don't know if it's valuable.


> @davorpa Isn't the copyright year 2007 instead of 2008?

Yes, you are right. I took it [from the dirlist](https://web.archive.org/web/20181010075503/https://www.bcl.hamilton.ie/~nmh/t3x.org/zzz/): `Lulu Press, 2018`. PDF is `2007`

Reply:
> I don't know if it's valuable.

The "R4RS Scheme interpreter" from 2009 is valuable, but could cause some confusion since it doesn't 100% match the code in the PDF. Suggestion: change its description to "R4RS interpreter (2009 update)".

Reply:
> Suggestion: change its description to "R4RS interpreter (2009 update)".

Done